Library/Homebrew/dev-cmd/edit.rb

```diff
@@ -62,6 +62,8 @@ def run
         exec_editor(*paths)
 
         if paths.any? do |path|
+             next if path.is_a?(String) && path == "--project"
```

Comment:
I don't think i understand the `String` check, i didn't think implicit conversion applied in equality checks:

```
$ brew irb
==> Interactive Homebrew Shell
Example commands available with: `brew irb --examples`
brew(main):001> Pathname("--project") == "--project"
=> false
brew(main):002> "--project" == Pathname("--project")
=> false
```

Reply:
`paths` is an `Array` of `Pathname`s or `String`s, see definition above (in particular L44).

Reply:
Ah, I guess you mean this could just be
```suggestion
             next if path == "--project"
```
?

Reply:
Yep, 'zactly, but I should've just included the suggested change for clarity. Thanks!